hack/verify-go-src.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
```

Comment:
It seems like this script runs all the other scripts. Do we need this?

Reply:
It runs all the scripts in `hack/go-tools/`, which right now is just the two being added in this PR. The exact arguments will be `./hack/verify-go-src.sh -v --rootdir $(pwd)`, which I can add as a new prow job. So we can have the existing prow job to verify all the generated code, and a new prow job to verify all the go source code. WDYT?

Reply:
OK, that seems reasonable to me.